### PR TITLE
Fix integer type of user ID

### PR DESCRIPTION
**Pull request by simonbrandhof**
Fix integer type of user ID

User IDs are INTEGER in database. The java classes are fixed
to replace long by int.

Note that this PR does not need to be fixed in 6.3.

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [GroupWsRef.java#L148](https://github.com/SonarSource/sonarqube/blob/4415ff811ab08c60faf6a461f998de603438f22d/server/sonar-server/src/main/java/org/sonar/server/usergroups/ws/GroupWsRef.java#L148): Remove this unnecessary cast to "int". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v62/PopulateOrganizationUuidOfGroups.java [hunk 1]
```diff
@@ -49,10 +49,10 @@ public void execute(Context context) throws SQLException {
     massUpdate.update("update groups set organization_uuid=?, updated_at=? where id=?");
     massUpdate.rowPluralName("groups");
     massUpdate.execute((row, update) -> {
-      long groupId = row.getLong(1);
+      int groupId = row.getInt(1);
```
Comment (sns-seb) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v62/PopulateOrganizationUuidOfGroups.java]: this is indeed correct (group id is an integer in DB) but this should either be a separate commit of the commit title should be updated

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v62/PopulateOrganizationUuidOfPermissionTemplates.java [hunk 2]
```diff
@@ -49,10 +49,10 @@ public void execute(Context context) throws SQLException {
     massUpdate.update("update permission_templates set organization_uuid=?, updated_at=? where id=?");
     massUpdate.rowPluralName("permission_templates");
     massUpdate.execute((row, update) -> {
-      long groupId = row.getLong(1);
+      int groupId = row.getInt(1);
```
Comment (sns-seb) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v62/PopulateOrganizationUuidOfPermissionTemplates.java]: same here

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v62/PopulateOrganizationUuidOfGroups.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v62/PopulateOrganizationUuidOfGroups.java]: and since you're at it, resource id also should be an integer (and the integer max value is going to kick us in the n.ts much faster on this one than on users or groups)

Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v62/PopulateOrganizationUuidOfGroups.java]: Ticket created for RESOURCE_ID: https://jira.sonarsource.com/browse/SONAR-8802

Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v62/PopulateOrganizationUuidOfGroups.java]: Commit title renamed.
